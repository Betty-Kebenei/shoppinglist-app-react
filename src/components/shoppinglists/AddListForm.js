import React, { Component } from 'react';
import instance from '../AxiosInstance';
import toastr from 'toastr';
import { 
    FormGroup, 
    ControlLabel,
    FormControl,
    HelpBlock,
    Button 
} from 'react-bootstrap';

class AddListForm extends Component {
    constructor(props) {
      super(props);
  
      this.handleChange = this.handleChange.bind(this);
  
      this.state = {
        listname: '',
        showModal3: false
      };
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }

    handleSubmit = (event) => {
        event.preventDefault();
        let values = new FormData();
        values.set("listname", this.state.listname);

        instance.post('/shoppinglists', values).then(
            response => {
                this.setState({listname:''});
                this.props.getAllShoppingLists();
                this.props.closeModal();
                toastr.success("You have successfully created a shopping lists!")
            }
        ).catch(error => {
            toastr.error(error.response.data.message)
        })
    }
  
    render() {
      return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <FormGroup
                        controlId="Listname"
                        >
                        <ControlLabel>Listname</ControlLabel>
                            <FormControl
                                type="text"
                                name="listname"
                                value={this.state.listname}
                                placeholder="Enter listname"
                                onChange={this.handleChange}
                                required
                            />
                    </FormGroup>
                    <Button type="submit">Submit</Button>
                    <FormControl.Feedback />
                </form>
            </div>
      );
    }
  }
  
export default AddListForm;