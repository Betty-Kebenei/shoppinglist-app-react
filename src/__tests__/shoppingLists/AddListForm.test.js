import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Config from '../../Config';

import AddListForm from '../../components/shoppinglists/AddListForm';

describe('interface of AddListForm component', () => {
    it('should render', () => {
        const wrapper = shallow(<AddListForm />);
        expect(wrapper).toMatchSnapshot();
    });
});