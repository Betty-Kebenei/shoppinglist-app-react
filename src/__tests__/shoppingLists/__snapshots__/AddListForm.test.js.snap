// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface of AddListForm component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddListForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={undefined}
      >
        <FormGroup
          bsClass="form-group"
          controlId="Listname"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Listname
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="listname"
            onChange={undefined}
            placeholder="Enter listname"
            required={true}
            type="text"
            value={undefined}
          />
        </FormGroup>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          type="submit"
        >
          Submit
        </Button>
        <FormControlFeedback
          bsClass="form-control-feedback"
          bsRole="feedback"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
            controlId="Listname"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Listname
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="listname"
              onChange={undefined}
              placeholder="Enter listname"
              required={true}
              type="text"
              value={undefined}
            />
          </FormGroup>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Submit
          </Button>,
          <FormControlFeedback
            bsClass="form-control-feedback"
            bsRole="feedback"
          />,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Listname
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="listname"
                onChange={undefined}
                placeholder="Enter listname"
                required={true}
                type="text"
                value={undefined}
              />,
            ],
            "controlId": "Listname",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Listname",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Listname",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "listname",
                "onChange": undefined,
                "placeholder": "Enter listname",
                "required": true,
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Submit",
            "disabled": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control-feedback",
            "bsRole": "feedback",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={undefined}
        >
          <FormGroup
            bsClass="form-group"
            controlId="Listname"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Listname
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="listname"
              onChange={undefined}
              placeholder="Enter listname"
              required={true}
              type="text"
              value={undefined}
            />
          </FormGroup>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Submit
          </Button>
          <FormControlFeedback
            bsClass="form-control-feedback"
            bsRole="feedback"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              controlId="Listname"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Listname
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="listname"
                onChange={undefined}
                placeholder="Enter listname"
                required={true}
                type="text"
                value={undefined}
              />
            </FormGroup>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              type="submit"
            >
              Submit
            </Button>,
            <FormControlFeedback
              bsClass="form-control-feedback"
              bsRole="feedback"
            />,
          ],
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Listname
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="listname"
                  onChange={undefined}
                  placeholder="Enter listname"
                  required={true}
                  type="text"
                  value={undefined}
                />,
              ],
              "controlId": "Listname",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Listname",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Listname",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "name": "listname",
                  "onChange": undefined,
                  "placeholder": "Enter listname",
                  "required": true,
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Submit",
              "disabled": false,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control-feedback",
              "bsRole": "feedback",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
